ui/Screen/Project/Commit.svelte

```diff
@@ -0,0 +1,131 @@
+<script>
+  import { Title, Flex, Code } from "../../DesignSystem/Primitive";
+  import { Icon } from "../../DesignSystem/Primitive";
+  import { gql } from "apollo-boost";
+  import { link } from "svelte-spa-router";
+  import { showNotification } from "../../store/notification.js";
+  import { getClient, query } from "svelte-apollo";
+  import { format } from "timeago.js";
+  import * as path from "../../lib/path.js";
+
+  export let params = null;
+  const projectId = params.id;
+  const commitHash = params.hash;
+
+  const QUERY = gql`
+    query($projectId: ID!, $commitHash: String!) {
+      commit(id: $projectId, sha1: $commitHash) {
+        author {
+          avatar
+          email
+          name
+        }
+        committerTime
+        message
+        sha1
+        summary
+      }
+    }
+  `;
+
+  async function fetchCommit() {
+    try {
+      const response = await query(getClient(), {
+        query: QUERY,
+        variables: {
+          projectId: projectId,
+          commitHash: commitHash
+        }
+      });
+      const result = await response.result();
+      const commit = result.data.commit;
+
+      // TODO(cloudhead): Fetch branch from backend.
+      commit.branch = "master";
+
+      return commit;
+    } catch (error) {
+      showNotification({
+        text: "Could not fetch commit",
+        level: "error"
+      });
+    }
+  }
+</script>
+
+<style>
+  header {
+    background: var(--color-foreground-level-1);
+    border-radius: 4px;
+    padding: 1.5rem;
+  }
+  .field {
+    color: var(--color-foreground-level-6);
+    margin-bottom: 0.5rem;
+  }
+  .field:last-child {
+    margin-bottom: 0;
+  }
+  .email {
+    font-family: var(--typeface-mono-regular);
+  }
+  .branch {
+    margin: 0 0.5rem;
+    font-weight: bold;
+    color: var(--color-foreground-level-6);
+  }
+  .author {
+    font-weight: bold;
+    color: var(--color-foreground);
+  }
+
+  /* TODO(cloudhead): These should be global */
+  a {
+    color: var(--color-secondary);
+  }
+  hr {
+    border: 0;
+    border-top: 1px solid var(--color-foreground-level-3);
+    margin: 1rem 0 1.5rem 0;
+  }
+</style>
+
+{#await fetchCommit() then commit}
+  <header>
+    <Flex align="space-between">
+      <Title variant="large" style="margin-bottom: 1rem">
+        {commit.summary}
+      </Title>
+      <span class="field">
+        Committed to
+        <span class="branch">
+          <Icon.Branch
+            color="foreground-level-6"
+            style="vertical-align: bottom" />
+          {commit.branch}
+        </span>
+        {format(commit.committerTime)}
+      </span>
+    </Flex>
+    <!-- TODO(cloudhead): Should be <pre> -->
+    <Code variant="code" style="margin-bottom: 1rem">{commit.message}</Code>
+    <hr />
+    <Flex align="space-between">
```

Comment:
Ah yes you might be right, I forgot about the slot thing.

Reply:
Ok I'm not sure this will work, the complication is that, if you look at the design file, we may sometimes get merge commit information, as well as commiter/author distinction, and want to bottom align in case there are two rows on one side and one row on the other side.

Reply:
Nevermind, got it working!

Reply:
083f443

Reply:
adding `style="align-items: flex-end;"` to the `Flex` in my code snippet, solves that problem.